samples/JobsAgent/JobsAgent.cpp

```diff
@@ -0,0 +1,291 @@
+                      << client_id << " Resubscribe Attempted. Result" << ResponseHelper::ToString(resubscribe_result)
+                      << std::endl
+                      << "*******************************************" << std::endl;
+            return ResponseCode::SUCCESS;
+        }
+
+        ResponseCode JobsAgent::Subscribe() {
+            AWS_LOG_INFO(LOG_TAG_JOBS_AGENT, "Subscribe");
+
+            mqtt::Subscription::ApplicationCallbackHandlerPtr p_next_handler =
+                std::bind(&JobsAgent::NextJobCallback, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
+
+            mqtt::Subscription::ApplicationCallbackHandlerPtr p_update_accepted_handler =
+                std::bind(&JobsAgent::UpdateAcceptedCallback, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
+
+            mqtt::Subscription::ApplicationCallbackHandlerPtr p_update_rejected_handler =
+                std::bind(&JobsAgent::UpdateRejectedCallback, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
+
+            util::Vector<std::shared_ptr<mqtt::Subscription>> topic_vector;
+            std::shared_ptr<mqtt::Subscription> p_subscription;
+
+            p_subscription = p_jobs_->CreateJobsSubscription(p_next_handler, nullptr, Jobs::JOB_DESCRIBE_TOPIC, Jobs::JOB_ACCEPTED_REPLY_TYPE, "$next");
+            topic_vector.push_back(p_subscription);
+
+            p_subscription = p_jobs_->CreateJobsSubscription(p_next_handler, nullptr, Jobs::JOB_NOTIFY_NEXT_TOPIC);
+            topic_vector.push_back(p_subscription);
+
+            p_subscription = p_jobs_->CreateJobsSubscription(p_update_accepted_handler, nullptr, Jobs::JOB_UPDATE_TOPIC, Jobs::JOB_ACCEPTED_REPLY_TYPE, "+");
+            topic_vector.push_back(p_subscription);
+
+            p_subscription = p_jobs_->CreateJobsSubscription(p_update_rejected_handler, nullptr, Jobs::JOB_UPDATE_TOPIC, Jobs::JOB_REJECTED_REPLY_TYPE, "+");
+            topic_vector.push_back(p_subscription);
+
+            ResponseCode rc = p_iot_client_->Subscribe(topic_vector, ConfigCommon::mqtt_command_timeout_);
+            return rc;
+        }
+
+        void JobsAgent::StartInstalledPackages() {
+            AWS_LOG_INFO(LOG_TAG_JOBS_AGENT, "StartInstalledPackages");
+
+            util::Map<util::String, util::String> statusDetailsMap;
+
+            for (util::JsonValue::ConstMemberIterator package_itr = installed_packages_json_.MemberBegin();
+                package_itr != installed_packages_json_.MemberEnd(); ++package_itr) {
+                if (package_itr->value.IsObject()) {
+                    util::String packageName = package_itr->name.GetString();
+                    if (PackageIsAutoStart(packageName.c_str())) {
+                        StartPackage(statusDetailsMap, packageName.c_str());
+                    }
+                }
+            }
+        }
+
+        ResponseCode JobsAgent::RunAgent(util::String processTitle) {
```

Comment:
you're immediately copying proccessTitle, why is it not 'const util::String &processTile'?